baseservice/iodevices/file_system/file_system_input_device.py

```diff
@@ -83,22 +97,26 @@ def read_file(device: 'FileSystemInputDevice',
 
         tmp_path = os.path.join(tmp_folder, get_random_id())
         try:
-            if not atomic_move(org_path, tmp_path,
-                               FileSystemInputTransaction._calc_lockfile_name(tmp_folder, org_path)):
-                return None, None
-        except AtomicMoveException as ex:
+            if not atomic_move(org_path,
```

Comment:
I think it would be more pythonic to have `atomic_move` raise an exception instead of returning bool, then use the relevant exception type to decide what handling would be most fitting for it.

Reply:
BTW i cant comment on the implementation of `atomic_move`, but check out `open`'s `opener` argument, it may simplify the implementation for you

Reply:
1. i don't really sure exceptions is better in this scenario.
2. i don't understand how "opener" helps me